optuna/integration/xgboost.py

```diff
@@ -43,20 +43,30 @@ class XGBoostPruningCallback(object):
             ``validation-merror``. Please refer to ``eval_metric`` in
             `XGBoost reference <https://xgboost.readthedocs.io/en/latest/parameter.html>`_
             for further details.
+        interval:
+            Check if trial should be pruned every n-th epoch. By default `interval=1` and
+            pruning is performed after every epoch. Increase `interval` to run several
+            epochs faster before applying pruning.
     """
 
-    def __init__(self, trial, observation_key):
-        # type: (optuna.trial.Trial, str) -> None
+    def __init__(self, trial, observation_key, interval=1):
+        # type: (optuna.trial.Trial, str, int) -> None
 
         _check_xgboost_availability()
 
         self._trial = trial
         self._observation_key = observation_key
+        self._interval = interval
 
     def __call__(self, env):
         # type: (xgb.core.CallbackEnv) -> None
 
         context = _get_callback_context(env)
+
+        # Iterations start with 1.
+        if (env.iteration + 1) % self._interval != 0:
```

Comment:
Are XGBoost users more familiar with logic that include iteration 0? E.g. https://github.com/dmlc/xgboost/blob/6a169cd41a5ea23b4371bfe2c1b39c6687bfebcd/python-package/xgboost/callback.py#L52-L53.

Reply:
Indeed, it is more reasonable to include iteration 0 to prune unpromising trial earlier. I changed it to `env.iteration % self._interval`.